### PR TITLE
Split into Tooltip and TooltipElement

### PR DESCRIPTION
**Pull request by sMteX**
I got it to work *somehow* with not rendering the wrapper element. In this version, 
```jsx
<Tooltip content="blah">
    <Link to="/premium">
        Link
     </Link>
</Tooltip>
```
renders
```html
<a href="/premium">Link</a>
```

I duplicated the original TooltipContainer class (naming is subject to change ofc) into AttachedTooltip class. Since now it's *supposed* to NOT render a wrapper element, I removed the `tagName, className, styles` props (as it shouldn't interfere with inner components, they can be styled directly).

### Important note about refs
As mentioned in the comment - the `AttachedTooltip` class should only have a single child component. This can be either directly an HTML element like `<div>` or another React Component like `<Link>`. Depending on the type, I either pass the `ref` as a `ref` property (to HTML elements) or `innerRef` property (to React components). But in order for the `innerRef` to work in inner React Components, the component must implement it itself (also spreading the other props, which are the events). 
```jsx
class SomeComponent extends React.Component {
  render() {
    const { innerRef, ...others } = this.props;
    return (<div ref={innerRef} {...others}>Hello</div>);
  }
}

<Tooltip content="Hello">
  <SomeComponent />
</Tooltip>
```

The `innerRef` must be specified explicitly in the `SomeComponent`, it doesn't get passed into `...others` because React seems to handle it differently. I thought this is somehow automatic in React, but it turns out `<Link>` actually does it manually: https://github.com/ReactTraining/react-router/blob/0853628daff26a809e5384f352fada57753fc1c3/packages/react-router-dom/modules/Link.js#L51

Also, currently the `React.Children.only()` also accepts React.Fragment as a valid child, but since it doesn't render, it doesn't do anything and just renders the contents without the tooltip. Not entirely desired maybe, but at least it doesn't seem to break anything.

It would be nice if we could refactor it some more, since a lot of the functionality is shared between the versions.

As for the main project, this comes with few changes:
1. I duplicated the Tooltip class into TooltipElement (naming is subject to change), again removed some props and instead of default export + named, I exported both as named:
```jsx
export { Tooltip, TooltipElement };
```
2. Renamed all usages of <Tooltip> to <TooltipElement> together with fixing the imports (if this PR passes, I'll update certain usages to <Tooltip> directly)

### CODE REVIEW

File: src/Tooltip.jsx [hunk 1]
```diff
@@ -47,82 +45,178 @@ class Tooltip extends React.PureComponent {
...
+            target={this.target.current}
+          />
+        )}
+      </React.Fragment>
     );
   }
 }
 
-export default Tooltip;
+const TooltipElement = ({ content, children, wrapperProps, ...others }) => (
```
Comment (MartijnHols) [inline src/Tooltip.jsx]: This should be in a separate file.

File: src/Tooltip.jsx [hunk 2]
```diff
@@ -1,44 +1,42 @@
...
-
-const stopProp = e => e.stopPropagation();
-
-class Tooltip extends React.PureComponent {
+class Tooltip extends React.Component {
   static propTypes = {
+    // eslint-disable-next-line react/no-unused-prop-types
     direction: PropTypes.string,
-    className: PropTypes.string,
+    tooltipClassName: PropTypes.string,
```
Comment (MartijnHols) [inline src/Tooltip.jsx]: We already call the component Tooltip, so when creating a Tooltip component instance and assigning props, I feel they shouldn't need to be prefixed.

File: src/Tooltip.jsx [hunk 3]
```diff
@@ -1,44 +1,42 @@
...
+import TooltipBubble from './TooltipBubble';
 
-// default colors
-const defaultColor = '#fff';
-const defaultBg = '#333';
-
-const stopProp = e => e.stopPropagation();
-
-class Tooltip extends React.PureComponent {
+class Tooltip extends React.Component {
```
Comment (MartijnHols) [inline src/Tooltip.jsx]:
```suggestion
class Tooltip extends React.PureComponent {
```

File: src/Tooltip.jsx [hunk 4]
```diff
@@ -1,44 +1,42 @@
...
+    // eslint-disable-next-line react/no-unused-prop-types
     direction: PropTypes.string,
-    className: PropTypes.string,
+    tooltipClassName: PropTypes.string,
     content: PropTypes.node.isRequired,
     background: PropTypes.string,
     color: PropTypes.string,
     padding: PropTypes.string,
+    eventOff: PropTypes.string,
+    eventOn: PropTypes.string,
```
Comment (MartijnHols) [inline src/Tooltip.jsx]: Probably makes more sense to place "on" before "off".

File: src/Tooltip.jsx [hunk 5]
```diff
@@ -1,44 +1,42 @@
...
   static defaultProps = {
     direction: 'up',
-    className: '',
+    tooltipClassName: '',
     background: '',
     color: '',
     padding: '10px',
     useHover: true,
     useDefaultStyles: false,
+    hoverDelay: 200,
```
Comment (MartijnHols) [inline src/Tooltip.jsx]: I think the default should be 0. The tooltips as they are feel kind of awkward.

File: src/Tooltip.jsx [hunk 6]
```diff
@@ -47,82 +45,178 @@ class Tooltip extends React.PureComponent {
...
-      portalProps.onTouchStart = stopProp;
+      // only use hover if they don't have a toggle event
+    } else if (useHover) {
+      props.onMouseOver = this.startHover;
+      props.onMouseOut = tipContentHover ? this.endHover : this.hideTip;
+      props.onTouchStart = this.toggleTip;
     }
 
+    // should ensure a single child (throws an error instead)
+    React.Children.only(children);
```
Comment (MartijnHols) [inline src/Tooltip.jsx]: I'd rather you did this check manually by verifying the children length. That would be much clearer code than this function call that acts like an assertion but is only obvious when reading the comment.

File: src/Tooltip.jsx [hunk 7]
```diff
@@ -1,35 +1,32 @@
...
-// default colors
-const defaultColor = '#fff';
-const defaultBg = '#333';
-
-const stopProp = e => e.stopPropagation();
+import TooltipBubble from './TooltipBubble';
 
 class Tooltip extends React.PureComponent {
   static propTypes = {
+    // eslint-disable-next-line react/no-unused-prop-types
```
Comment (MartijnHols) [inline src/Tooltip.jsx]: What is unused? Why? Can we drop this comment?

Comment (sMteX) [reply src/Tooltip.jsx]: I'll try to look - I'd want to document the properties somewhere anyway, where would you suggest? Inline here with comments or revise/update `README.md`? Also, if possible, after we're done, I'd update the `index.d.ts` as well

**Pull request merged**